### PR TITLE
boxel-motion improvements

### PR DESCRIPTION
**Pull request by lukemelia**
- Add Split View example to boxel-motion test app as a study for the AI Assistant Panel animation
- Incorporate boxel-motion builds into CI
- Generalize fill behavior of behaviors rather than special casing StaticBehavior treatment
- Default StaticBehavior to a duration of 1, since you can now opt into forward fill
    - when using StaticBehavior with SpringBehavior, you don't know how long the animation will be, and you may want a static animation to persist for the duration of the animation
- Use context-relative bounds for removed sprites

_extracted from #1097_

### CODE REVIEW

**Comment by lukemelia** [comment]
https://github.com/cardstack/boxel/assets/353/afc1a59b-e4a9-47c4-bce9-5c3f92437552

File: QUICKSTART.md [hunk 1]
```diff
@@ -21,12 +21,14 @@ To build the entire repository and run the application, follow these steps:
    pnpm install
    ```
 
-4. Build the boxel-ui:
+4. Build the boxel-ui and boxl-motion addons:
```
Comment (backspace) [inline QUICKSTART.md:24]:
```suggestion
4. Build the boxel-ui and boxel-motion addons:
```

File: README.md [hunk 2]
```diff
@@ -45,8 +45,9 @@ Make sure that you have created a matrix user for the base realm, drafts realm,
 In order to run the ember-cli hosted app:
 
 1. `pnpm build` in the boxel-ui/addon workspace to build the boxel-ui addon.
-2. `pnpm start` in the host/ workspace to serve the ember app. Note that this script includes the environment variable `OWN_REALM_URL=http://localhost:4201/draft/` which configures the host to point to the draft realm's cards realm by default.
-3. `pnpm start:all` in the realm-server/ to serve the base realm, draft realm and published realm -- this will also allow you to switch between the app and the tests without having to restart servers)
+2. `pnpm build` in the boxel-motion/addon workspace to build the boxel-motion addon.
+3. `pnpm start` in the host/ workspace to serve the ember app. Note that this script includes the environment variable `OWN_REALM_URL=http://localhost:4201/draft/` which configures the host to point to the draft realm's cards realm by default.
```
Comment (backspace) [inline README.md:47]:
```suggestion
3. `pnpm start` in the host/ workspace to serve the ember app. Note that this script includes the environment variable `OWN_REALM_URL=http://localhost:4201/drafts/` which configures the host to point to the draft realm's cards realm by default.
```

I know it was `/draft/` already but that’s wrong right?

File: packages/boxel-motion/addon/src/components/animation-context.gts [hunk 3]
```diff
@@ -67,6 +76,17 @@ export default class AnimationContextComponent
...
+    super(owner, args);
+    if (!this.animations) {
+      throw new Error(
+        `Expected to find "animations" service in app.
+         Add 'app/services/animations.ts' with
+           \`export { AnimationsService as default } from '@cardstack/boxel-motion';\``,
+      );
+    }
+  }
+
```
Comment (backspace) [inline packages/boxel-motion/addon/src/components/animation-context.gts:89]: Helpful 🎉

Comment (backspace) [inline packages/boxel-motion/addon/src/components/animation-context.gts:89]: Helpful 🎉

File: packages/boxel-motion/test-app/app/templates/application.hbs [hunk 4]
```diff
@@ -39,7 +39,7 @@
   <li>
     <LinkTo @route="removed-sprite-interruption">
       Removed Sprite Interruption
-    </LinkTo>
+    </LinkTo>
```
Comment (backspace) [inline packages/boxel-motion/test-app/app/templates/application.hbs:42]:
```suggestion
    </LinkTo>
```

what is this 😯

**Review by backspace — APPROVED** [review]
`@debugging` is great 🤩

**Comment by lukemelia** [comment]
A few other thoughts, so I don't forget them:

1) Specifying `to` and `from` can be confusing. The more common case appears to be using captured to and/or from in combination with an "entrance" value for an Inserted Sprite and an "exit" value for a Removed Sprite
2) the strategy of cloning destroyed sprites to the orphans div under the animation context can cause problems for nested sprites. Given the structure AnimationContext > Sprite A > Sprite B... for animating Sprite B, Inserted and Kept animations are run with a parent of Sprite A, but Removed animations are run with a parent of the AnimationContext. Perhaps it should be cloned beneath Sprite A instead when Removed?

**Review by ef4 — APPROVED** [review]
This seems fine as an incremental step. But in terms of the vision for how I think boxel-motion is supposed to work, `fill` doesn't really make sense.

The use case for boxel-motion is that you already have an initial and final DOM and the behaviors explain what happens in between those states. If behaviors are able to alter the final DOM, it seems like something about the architecture is getting muddied.

**Pull request merged**